src/lib.rs

```diff
@@ -130,6 +126,13 @@ pub mod poll;
 mod config;
 mod error;
 
+/// The set of requirements for watcher event handling functions.
+pub trait EventFn: 'static + Fn(Result<Event>) + Send + Sync {}
+
+impl<F> EventFn for F
+where
+    F: 'static + Fn(Result<Event>) + Send + Sync {}
```

Comment:
I'd use something like `type OnEvent = dyn Fn(Result<Event>) + Send + Sync + 'static`

Reply:
Thanks for the tip, but does this matter much? Are there bigger motivations for using the type alias, or is it mostly just to avoid having to prefix `dyn` everywhere? I'm currently leaning towards leaving this as it is, but want to check I'm not missing some subtly incurred cost by leaving it this way.

Reply:
It’s not to avoid dyn, it’s to avoid an extra trait and just lean on existing Fn